File: Python/shutdown.py
from os import system, name
from time import sleep

if name == "posix":
    shutdown = "shutdown -h +"
    cancel = "shutdown -c"
    multiply = 1
elif name == "nt":
    shutdown = "shutdown /s /t "
    cancel = "shutdown /a"
    multiply = 60
else:
    exit()

while 1:
    system("cls||clear")
    menu = input("1- Set the shutdown time\n2- Cancel the shutdown\n3- Exit\n\n>>>")
    system("cls||clear")
    if menu == "1":
        try:
            time = int(input("How many minutes later do you want the computer to shutdown: "))
        except ValueError:
            print("\nPlease type number..")
            sleep(1.5)
            continue
        system(shutdown+str(time*multiply))
    elif menu == "2":
        system(cancel)
        print("The shutdown has been canceled..")
        sleep(1.5)
    elif menu == "3":
        break
    else:
        print("Please make a valid choice..")
        sleep(1.5)

File: Haskell/shutdown.hs
import System.Info (os)
import System.IO (stdout, hFlush)
import Control.Monad (forever)
import System.Process (callCommand)
import Control.Concurrent (threadDelay)
import System.Exit (exitSuccess)

main :: IO ()
main = do
    let (shutdown, cancel, multiply) = 
            if os == "linux" then
                ("shutdown -h +", "shutdown -c", 1)
            else if os == "mingw32" || os == "cygwin" || os == "windows" then
                ("shutdown /s /t ", "shutdown /a", 60)
            else
                ("", "", 0)

    forever $ do
        callCommand "cls||clear"
        putStr"1- Set the shutdown time\n2- Cancel the shutdown\n3- Exit\n\n>>>"
        hFlush stdout
        menu <- getLine
        callCommand "cls||clear"
        if menu == "1" then do
            putStr "How many minutes later do you want the computer to shutdown: "
            hFlush stdout
            timeStr <- getLine
            let time_ = reads timeStr :: [(Int, String)]
            case time_ of
                [(timeInt, "")] -> do 
                    
                    callCommand("cls||clear && "++shutdown++show(timeInt*multiply))
                    threadDelay 3000000
                _ -> do
                    putStrLn "\nPlease type number.."
                    threadDelay 2000000
        else if menu == "2" then do
            callCommand cancel
            putStrLn "The shutdown has been canceled.."
            threadDelay 2000000
        else if menu == "3" then do
            exitSuccess
        else do
            putStrLn "Please make a valid choice.."
            threadDelay 2000000